tests/js/client/shell/aql-failures-cluster.js

```diff
@@ -0,0 +1,158 @@
+    return 'http://' + endpoint;
+  }
+  return 'http' + endpoint.substr(pos);
+};
+
+const getEndpointById = (id) => {
+  const toEndpoint = (d) => (d.endpoint);
+  const instanceInfo = JSON.parse(internal.env.INSTANCEINFO);
+  return instanceInfo.arangods.filter((d) => (d.id === id))
+    .map(toEndpoint)
+    .map(endpointToURL)[0];
+};
+
+const callFinish = (server, route) => {
+  const res = request.delete({
+    url: getEndpointById(server) + "/" + route,
+    body: { code: 0 }
+  });
+  // Did not leave an engine behind
+  assertEqual(res.status, 200);
+};
+
+
+function aqlFailureSuite () {
+  'use strict';
+  const cn = "UnitTestsAhuacatlFailures";
+
+  let  assertFailingQuery = function (query, error) {
+    if (error === undefined) {
+      error = internal.errors.ERROR_DEBUG;
+    }
+    try {
+      db._query(query);
+      fail();
+    } catch (err) {
+      assertEqual(error.code, err.errorNum, query);
+    }
+  };
+
+  const cleanUpLeftovers = (warnings) => {
+    for (const {code, message} of warnings) {
+      if (code === ERROR_CLUSTER_AQL_COMMUNICATION.code) {
+        const parts = message.split(" ");
+        const route = parts.filter(m => m.startsWith("_api/aql"))[0];
+        assertTrue(route !== undefined);
+        const serverInfo = parts.filter(m => m.startsWith("server:"))[0];
+        assertTrue(serverInfo !== undefined);
+        const serverName = serverInfo.split(":")[1];
+        assertTrue(serverName !== undefined);
+        callFinish(serverName, route);
+      }
+    }
+  };
+        
+  return {
+    setUpAll: function () {
+      internal.debugClearFailAt();
+      db._drop(cn);
+      db._create(cn, { numberOfShards: 3 });
+    },
+
+    setUp: function () {
+      internal.debugClearFailAt();
+    },
+
+    tearDownAll: function () {
+      internal.debugClearFailAt();
+      db._drop(cn);
+    },
+
+    tearDown: function () {
+      internal.debugClearFailAt();
+    },
+
+    testThatQueryIsntStuckAtShutdownIfFinishDBServerPartsThrows: function() {
+      // Force cleanup to fail.
+      // This should result in a positive query, but may leave locks on DBServers.
+      internal.debugSetFailAt("Query::finalize_error_on_finish_db_servers");
+      // NOTE we need to insert two documents here to not end up with a single remote operation node.
+      const {stats, warnings} = db._query(`FOR value IN 1..2 INSERT {value} INTO ${cn}`).getExtra();
+      try {
+        // We force a failure upon aql/finish which produces the statistics. They are expected to be missing now.
+        assertEqual(stats.writesExecuted, 0);
+        // We should have at least one warning due to a failed finish call.
+        assertTrue(warnings.length > 0);
+      } finally {
+        cleanUpLeftovers(warnings);
+      }
+    },
+
+    /*
+     * This test is disabled due to the following reasons:
+     * If we trigger an error during the commit, the warnings are removed from the query, it is repliced by an error only.
```

Comment:
What does `repliced` mean here? Replaced, replicated?

Reply:
replaced, fixed it.